### PR TITLE
[wasm] [debugger] Add test suite.

### PR DESCRIPTION
**Pull request by thaystg**
Based on what Kumpera have done here:
https://github.com/mono/mono/pull/11310

### CODE REVIEW

File: sdks/wasm/ProxyDriver/TestHardnessStartup.cs [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.AspNetCore.StaticFiles;
+using Newtonsoft.Json.Linq;
+using WsProxy;
+
+
+namespace WsProxy {
+	public class TestHarnessStartup {
```
Comment (lewing) [inline sdks/wasm/ProxyDriver/TestHardnessStartup.cs]: Looks like there is a typo in the filename

File: sdks/wasm/ProxyDriver/TestHardnessStartup.cs [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+			if (configuration ["NodeApp"] != null) {
+				var nodeApp = configuration ["NodeApp"];
+				Console.WriteLine($"Doing the nodejs: {nodeApp}");
+				Console.WriteLine (Path.GetFullPath (nodeApp));
+				var nodeFullPath = Path.GetFullPath (nodeApp);
+				psi.Arguments = $"--inspect-brk=localhost:0 {nodeFullPath}";
+				psi.FileName = "node";
+
+				app.UseRouter (router => {
```
Comment (lewing) [inline sdks/wasm/ProxyDriver/TestHardnessStartup.cs]: the NodeList logit is similar to the UseDebugProxy logic in Startup.cs we should probably unify the code in a way that makes sense

File: sdks/wasm/Mono.WebAssembly.DebuggerProxy/WsClient.cs [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+				current_write = socket.SendAsync (new ArraySegment<byte> (pending_writes [0]), WebSocketMessageType.Text, true, token);
+				return current_write;
+			}
+			return null;
+		}
+
+		async Task<string> ReadOne (CancellationToken token)
+		{
+			byte [] buff = new byte [4000];
+var mem = new MemoryStream ();
```
Comment (lewing) [inline sdks/wasm/Mono.WebAssembly.DebuggerProxy/WsClient.cs]: indentation

File: sdks/wasm/ProxyDriver/TestHardnessStartup.cs [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+
+		async Task SendNodeList (HttpContext context)
+		{
+			Console.WriteLine ("hello chrome! json/list");
+			try {
+				var response = new JArray (JObject.FromObject (new {
+					description = "node.js instance",
+					devtoolsFrontendUrl = "chrome-devtools://devtools/bundled/inspector.html?experiments=true&v8only=true&ws=localhost:9300/91d87807-8a81-4f49-878c-a5604103b0a4",
+					faviconUrl = "https://nodejs.org/static/favicon.ico",
+id = "91d87807-8a81-4f49-878c-a5604103b0a4",
```
Comment (lewing) [inline sdks/wasm/ProxyDriver/TestHardnessStartup.cs]: indentation

File: sdks/wasm/ProxyDriver/TestHardnessStartup.cs (hunk 2 above)
Comment (thaystg) [reply sdks/wasm/ProxyDriver/TestHardnessStartup.cs]: OK, I will think about that and try to change it in my next PR. :)

**Comment by thaystg** [comment]
@monojenkins build failed

**Comment by thaystg** [comment]
@monojenkins build failed

**Comment by thaystg** [comment]
@monojenkins build failed

**Comment by luhenry** [comment]
@thaystg there is a conflict, so you will need to resolve it before we can merge, and that's going to trigger a new build anyway.

File: scripts/ci/run-jenkins.sh [hunk 5]
```diff
@@ -255,20 +255,20 @@ if [[ ${CI_TAGS} == *'webassembly'* ]] || [[ ${CI_TAGS} == *'wasm'* ]];
             ${TESTCMD} --label=sm-mini-test --timeout=60m $gnumake -C sdks/wasm run-sm-mini
             ${TESTCMD} --label=jsc-mini-test --timeout=60m $gnumake -C sdks/wasm run-jsc-mini
             #The following tests are not passing yet, so enabling them would make us perma-red
-            #${TESTCMD} --label=mini-corlib --timeout=60m make -C sdks/wasm run-all-corlib
-            #${TESTCMD} --label=mini-system --timeout=60m make -C sdks/wasm run-all-system
-            ${TESTCMD} --label=ch-system-core --timeout=60m make -C sdks/wasm run-ch-system-core
-            ${TESTCMD} --label=v8-system-core --timeout=60m make -C sdks/wasm run-v8-system-core
-            ${TESTCMD} --label=sm-system-core --timeout=60m make -C sdks/wasm run-sm-system-core
-            ${TESTCMD} --label=jsc-system-core --timeout=60m make -C sdks/wasm run-jsc-system-core
+            #${TESTCMD} --label=mini-corlib --timeout=60m $gnu$gnumake -C sdks/wasm run-all-corlib
```
Comment (luhenry) [inline scripts/ci/run-jenkins.sh]: That's a typo `$gnu$gnumake` -> `$gnumake`

**Comment by thaystg** [comment]
@monojenkins build failed

**Comment by thaystg** [comment]
@monojenkins build failed

**Comment by luhenry** [comment]
@monojenkins squash

**Pull request merged**